### PR TITLE
Feat: Singlevideo, like, history

### PR DESCRIPTION
**Pull request by yashpatel1797**
Feat added:
1. Single video page
2. like/ dislike button
3. like videos page
4. history page

### CODE REVIEW

**Review by sonishreyas — APPROVED** [review]
The code looks clean and easy to understand. At some places you might want to use preetier. Rest it looks good

**Review by sharmayatendra — APPROVED** [review]
it's good that you have used JS docs while writing code
good use of CSS module
good file structure

File: src/pages/VideoPage/VideoPage.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            removeVideoFromWatchLater(video._id, playListDispatch);
+    }
+    const addWatchListHandler = () => {
+        !isLogin ? navigate("/login") :
+            addVideoInWatchLater(video, playListDispatch);
+    }
+    const saveToPlaylistHandler = () => {
+        !isLogin ? navigate("/login") :
+            setShowModal(true);
+    }
```
Comment (devdeadviz) [inline src/pages/VideoPage/VideoPage.js:42]: You can extract these functions as well to make them reusable.

File: src/utilities/services/history-helper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * @param {*} playListDispatch 
+ */
+const addVideoInHistory = async (video, playListDispatch) => {
+    try {
+        const { data } = await axios.post("/api/user/history", { video }, {
+            headers: { authorization: encodedToken }
+        })
+        playListDispatch({ type: "ADD_HISTORY", payload: data.history })
+    } catch (error) {
+        console.log(error);
```
Comment (devdeadviz) [inline src/utilities/services/history-helper.js:15]: Instead of consoling the error, you can use `toastify` to show pop up whenever some error occurs.

**Review by devdeadviz — APPROVED** [review]
The code is clean, has a good file structure, you have followed all the best practices. Gook work!

**Pull request merged**